src/codegen_arc.cpp

```
@@ -193,18 +194,17 @@ llvm::FunctionCallee CodeGen::resolveDestructor(llvm::AllocaInst *alloca) {
     return {};
 }
 
-CodeGen::ResourceKind CodeGen::detectResourceKind(llvm::Value *val) {
-    for (int i = 0; i < RK_COUNT; ++i)
+int CodeGen::detectResourceKind(llvm::Value *val) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)